pkgs/development/compilers/graalvm/graalvm-oracle/default.nix

```diff
@@ -3,14 +3,21 @@
   stdenv,
   fetchurl,
   graalvmPackages,
+  onnxruntime,
   useMusl ? false,
-  version ? "23",
+  version ? "24",
 }:
 
-graalvmPackages.buildGraalvm {
+let graal = (graalvmPackages.buildGraalvm {
   inherit useMusl version;
   src = fetchurl (import ./hashes.nix).${version}.${stdenv.system};
   meta.platforms = builtins.attrNames (import ./hashes.nix).${version};
   meta.license = lib.licenses.unfree;
   pname = "graalvm-oracle";
-}
+}).overrideAttrs (prev: {
+  propagatedBuildInputs = (prev.propagatedBuildInputs or []) ++ [ onnxruntime ];
+  postFixup = (prev.postFixup or "") + ''
+    patchelf --replace-needed libonnxruntime.so.1.18.0 libonnxruntime.so.1 $out/lib/svm/profile_inference/onnx/native/libonnxruntime4j_jni.so
+  '';
```

Comment:
How does this differ from the `autoPatchelfIgnoreMissingDeps` approach used in PR #423088 ?

Reply:
@emaryn I think your version of this PR was probably best, it's too bad you closed it. I was mostly just trying to get it to work first and what you reviewed was not a cleaned u pversion, but I agree that patchelf step should only be done on 24 onwards.

I think the issue requesting an update to the package and our PRs all landed in quick succession, I was not aware of either that issue or your PR when I worked on this.

As for keeping older versions around, I'm not sure what the best policy is, maintaining these versions is probably not a big deal (until the build system changes but I think that's unlikely), but a warning about EOL would be nice to have.